core/src/services/routing.js

```diff
@@ -87,6 +87,9 @@ const removeElementChildren = node => {
 export const isNotSameDomain = (config, component) => {
   if (config.iframe) {
     const componentData = component.get();
+    if (componentData.isNavigateBack) {
```

Comment:
I wouldnt change the logic here. The fact that we are in the middle of back navigation doesnt make suddenly that source and target url belong  to the same origin.

IMHO this logic should be unchanged, but we should rather mainpulate the if statement in the navigateIframe function

sth like this
```
    if(!componentData.isNavigateBack &&
    (isNotSameDomain(config, component) ||
    hasIframeIsolation(component) ||
    Boolean(config.builderCompatibilityMode))
{
...
}
```